docs/how_to/use_toggles.rst

```diff
@@ -0,0 +1,65 @@
+***************************************
+How to: Use toggles
+***************************************
+
+Calling toggles
+==================
+
+Unlike the underlying waffle library which looks up waffles by name,
+toggles should be imported and used directly:
+
+.. code:: python
+
+    from whereever import FLAG_FOR_STUFF, COURSE_FLAG_FOR_STUFF
+
+    FLAG_FOR_STUFF.is_enabled()
+    COURSE_FLAG_FOR_STUFF.is_enabled('course-key-123')
+
+You can get at the state of these toggles by name via the waffle library
+and it will appear to work fine in simple testing. Don't be fooled.
+
+How toggles get their results
+====================================
+
+Obviously since some toggles take into account the user they must have more data
+than the zero arguments of ``is_enabled()``.
+Under the hood toggles use:
+
+* A per-request cache, so a toggle value is stable during the request
+* If needed by this toggle type, the request via ``crum.get_request`` to pass to the underlying waffle code
+
+
+Using toggles you don't want to import
+==========================================
+
+If you are importing a dependency you can directly check its toggles as
+you can your own toggles. But if you want to have an indirect interaction
+(such as getting an object you do not import via dependency injection),
+that's off the table.
+
+As mentioned above you could go directly to the underlying waffle check but should not.
+Best practice is to expose the interesting boolean as an API call itself.
+
+So in the source that you don't want to import:
+
+.. code:: python
+
+    from edx_toggles.toggles import WaffleFlag
+
+    INTERESTING_FLAG = WaffleFlag('dangerous_library.interesting_flag', __name__)
+
+    # probably in a separate API file
+    def is_interesting_on():
+        return INTERESTING_FLAG.is_enabled()
+
+And then in the code which is interested but avoiding a full import:
+
+.. code:: python
+
+    interesting_service = inject_me_a_service('interesting')
+
+    if interesting_service.is_interesting_on:
+        do_the_thing(interesting_service)
+
+The toggle will pick up whatever request cache and request is currently
```

Comment:
maybe just say several of the toggles pick up the request. I think all of them may touch it just because of the cache, but only some of them are interested in request values. I'll check.

I think knowing most of your toggles are dependent on requests is important (for example that's why they're not going to work in a celery task as was recently discovered in a team project)

Reply:
I would be specific. The other doc is broken up by toggle type, so docs could stay relevant to the type. You are really just talking the about WaffleFlag (and any related toggles like CourseWaffleFlag, etc.) that is dependent on the cache.

Note that WaffleSwitch might also have its result cached (which may be useful to know?), but should work fine and just not be cached when there is no request.